File: src/services/get_def/get_def_request.ml
(*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *)

type ('M, 'T) member_info = {
  prop_name: string;
  object_type: 'T;
  force_instance: bool;
}

type ('M, 'T) t =
  | Identifier of {
      name: string;
      loc: 'T;
    }
  | Type of 'T
  | Typeof of 'T
  | Member of ('M, 'T) member_info
  | JsxAttribute of {
      component_t: 'T;
      name: string;
      loc: 'M;
    }

File: rust_port/crates/flow_services_get_def/src/get_def_request.rs
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

use flow_data_structure_wrapper::smol_str::FlowSmolStr;

#[derive(Debug, Clone)]
pub struct MemberInfo<T> {
    pub prop_name: FlowSmolStr,
    pub object_type: T,
    pub force_instance: bool,
}

#[derive(Debug, Clone)]
pub enum GetDefRequest<M, T> {
    Identifier {
        name: FlowSmolStr,
        loc: M,
    },
    Member(MemberInfo<T>),
    JsxAttribute {
        component_t: T,
        name: FlowSmolStr,
        loc: M,
    },
}
